### DISCUSSION TITLE
Does Prisma Client not exports all types like XCreateInput?

### DISCUSSION THREAD

**Discussion by Vegetappen** [prisma/prisma#12543] [Q&A] [answered]
I'm playing around the basic example, I find 
```typescript
export type PostCreateInput = {
  createdAt?: Date | string;
  updatedAt?: Date | string;
  title: string;
  content?: string | null;
};
```
in the generated Prisma Client.
But I cannot import that `PostCreateInput` from `@prisma/client`, the client do exports type `Post`.
Is it designed on purpose? Do I have to recode something like `class MyPostInput implements Partial<Post>`? Really need advice.

**Comment by austincrim** [comment, marked as answer]
Hey @hazzard1014 👋 ,

Those types are namespaced under the `Prisma` export.
```ts
import { Prisma } from '@prisma/client'

const myInput: Prisma.PostCreateInput
```

Let me know if that works!

**Reply by Vegetappen** [reply]
Thanks, that works.

**Comment by steliosrammos** [comment]
Hi @austincrim I am getting a similar issue and going through `Prisma` does not work for me either. Any idea why?

I am running Prisma version 3.11.0.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
